packages/frontend/src/providers/DashboardProvider.tsx

```diff
@@ -420,17 +424,29 @@ export const DashboardProvider: React.FC = ({ children }) => {
     const addSuggestions = useCallback(
         (newSuggestionsMap: Record<string, string[]>) => {
             setFieldsWithSuggestions((prev) => {
-                return Object.entries(prev).reduce<FieldsWithSuggestions>(
-                    (sum, [key, field]) => {
-                        const currentSuggestions = field?.suggestions || [];
-                        const newSuggestions = newSuggestionsMap[key] || [];
+                const updatedFields: FieldsWithSuggestions = {};
+                for (const key in prev) {
+                    if (prev.hasOwnProperty(key)) {
+                        const field = prev[key];
+
+                        const currentSuggestions = field?.suggestions ?? [];
+
+                        const newSuggestions =
+                            newSuggestionsMap[key] || new Set();
                         const suggestions = Array.from(
-                            new Set([...currentSuggestions, ...newSuggestions]),
+                            new Set([
+                                ...currentSuggestions,
+                                ...Array.from(newSuggestions),
```

Comment:
How many of these do we think there are? Like how many things in this loop? If its just a few this might be ok, but if there are thousands I think this is going to be slow. We create a new set on line 435, then allocate memory for a new array on 436, then a new set on 437, then a new array on 439. I think we could tighten this up an have fewer allocations. I'm happy to look closer if you want. 

I'm not trying to be too nit-picky! We are doing perf work or I wouldn't notice so much 😄

Reply:
Removing suggestions https://github.com/lightdash/lightdash/pull/7942/commits/ae13563c187da046e8f97a56ff124f25315a2c5f ! Testing this PR + testing on `main` and will chat with @IrakliJani on the consequences of this
But will address this spreading and other operations that would be heavy if we consider re-adding these `fieldsWithSuggestions`, thanks 👍